htdocs/admin/content/messages.php

```diff
@@ -1,822 +1,714 @@
-			<input type=\"radio\" name=\"message_massmail\" value=\"1\" /> Ja</td></tr>";
-			echo "<tr><th>Gel&ouml;scht</th><td><input type=\"radio\" name=\"message_deleted\" value=\"2\" checked=\"checked\" /> Egal
-			<input type=\"radio\" name=\"message_deleted\" value=\"0\" /> Nein
-			<input type=\"radio\" name=\"message_deleted\" value=\"1\" /> Ja</td></tr>";
-			echo "<tr><th>Kategorie</th><td><select name=\"message_cat_id\">";
-			echo "<option value=\"\">(egal)</option>";
-			$cres = dbquery("SELECT cat_id,cat_name FROM message_cat ORDER BY cat_order;");
-			while ($carr = mysql_fetch_array($cres))
-			{
-				echo "<option value=\"".$carr['cat_id']."\">".$carr['cat_name']."</option>";
-			}
-			echo "</select></tr>";
-			echo "<tr><th>Anzahl Datens&auml;tze</th><td class=\"tbldata\"><select name=\"message_limit\">";
-			for ($x=100;$x<=2000;$x+=100)
-				echo "<option value=\"$x\">$x</option>";
-			echo "</select></td></tr>";
-
-			echo "</table>";
-			echo "<br/><input type=\"submit\" class=\"button\" name=\"user_search\" value=\"Suche starten\" /></form>";
-
-			$tblcnt = mysql_fetch_row(dbquery("SELECT count(*) FROM messages;"));
-			echo "<br/>Es sind ".nf($tblcnt[0])." Eintr&auml;ge in der Datenbank vorhanden.";
-		}
-
-	}
-?>
-
+use EtoA\Admin\AdminUser;
+use EtoA\Message\MessageCategoryRepository;
+use EtoA\Message\MessageRepository;
+use EtoA\Message\ReportRepository;
+use EtoA\User\UserRepository;
+use Symfony\Component\HttpFoundation\Request;
+
+define("USER_MESSAGE_CAT_ID",1);
+define("SYS_MESSAGE_CAT_ID",5);
+
+/** @var \EtoA\User\UserRepository $userRepository */
+$userRepository = $app['etoa.user.repository'];
+
+/** @var MessageRepository */
+$messageRepository = $app[MessageRepository::class];
+
+/** @var MessageCategoryRepository */
+$messageCategoryRepository = $app[MessageCategoryRepository::class];
+
+/** @var ReportRepository */
+$reportRepository = $app[ReportRepository::class];
+
+/** @var Request */
+$request = Request::createFromGlobals();
+
+if ($sub=="sendmsg") {
+    sendMessageForm($request, $cu, $messageRepository, $userRepository);
+} elseif ($sub=="reports") {
+    manageReports($request, $reportRepository, $userRepository);
+} else {
+    manageMessages($request, $messageRepository, $messageCategoryRepository, $userRepository);
+}
+
+function sendMessageForm(
+    Request $request,
+    AdminUser $cu,
+    MessageRepository $messageRepository,
+    UserRepository $userRepository
+): void {
+    global $page;
+    global $sub;
+
+    echo "<h1>Nachrichten</h1>";
+
+    echo "Nachricht an einen Spieler senden:<br/><br/>";
+
+    $subj = $request->query->get('message_subject', '');
+    $text = "";
+    if ($request->request->has('submit'))
+    {
+        if ($request->request->get('message_subject') != "" && $request->request->get('message_text') != "")
+        {
+            $to = [];
+            if ($request->request->getInt('rcpt_type') == 1) {
```

Comment:
===

Reply:
Its an int to int comparison, is it really needed to ensure strict comparison? then we have to change it in maaany places....

Reply:
Ok havent found anything in the PSRs, but symfony coding standard says strict comparison should always used. 
https://symfony.com/doc/current/contributing/code/standards.html#structure

If this is the one we like to use, then we should go for that.

Reply:
yes, agree that in the long run strict comparison should always be used. This is going to be easier to do the higher the phpstan level we are using. I think we can change this in a later iteration too